swan-lake/learn-the-platform/test-document-the-code/test-ballerina-code/test-services-and-clients.md

```diff
@@ -110,6 +110,31 @@ function getMockResponse() returns http:Response {
 }
 ```
 
+### Mock `final` clients
+
+Object mocking cannot be used as final clients cannot be modified. To facilitate testing, it is recommended 
+to write the client initialization logic in a separate function and assign the returned value to the client. 
+This initialization function can then be mocked using the legacy function mocking feature.
+
+***Example:***
+The following is a simple example on how to mock a `final` client.
+
+Initialize the client:
+```bal
+final http:Client clientEndpoint = check intializeClient();
```

Comment:
Shall we include the import statements as well and make the code complete? We have followed same approach for other sections as well.

Reply:
Added with 894fa70